ui/src/main/webapp/src/app/project/migration-project-form.component.ts

```diff
@@ -49,6 +50,10 @@ export class MigrationProjectFormComponent extends FormComponent implements OnIn
             }
 
             this.isInWizard = flatRouteData.data.hasOwnProperty('wizard') && flatRouteData.data['wizard'];
+            this.isCreating = flatRouteData.data.hasOwnProperty('creating') && flatRouteData.data['creating'];
+
+            if(this.isCreating)
+                this._migrationProjectService.deleteProvisionalProjects().subscribe(res => {});
```

Comment:
@OndraZizka - Yeah, a debounce or something would be nice. It doesn't flicker on master, though, so maybe you have changed something?

I think you are right that a race condition really can't happen here.